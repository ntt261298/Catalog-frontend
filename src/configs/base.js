export default {
  apiHost: 'http://localhost:5000',
};
