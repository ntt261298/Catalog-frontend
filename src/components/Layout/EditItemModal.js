import React from 'react';
import { connect } from 'react-redux';
import {
  Button, Modal, ModalHeader, ModalBody, ModalFooter,
} from 'reactstrap';
import { PropTypes } from 'prop-types';
import { hideModal } from 'actions/app';
import getCategories from 'actions/category';
import { editItem, getItems, getUserItems } from 'actions/item';
import { selectAllCategories } from 'utils/selector';
import { errMessage, successMessage } from 'utils/messages';


export class EditItemModal extends React.Component {
    state = {
      title: '',
      description: '',
      categoryID: '',
      itemID: '',
    }

    componentDidMount() {
      this.props.getCategories();
      this.setState(this.props.item);
    }

    changeInput = (e) => {
      this.setState({
        [e.target.name]: e.target.value,
      });
    }

    editItem = async () => {
      const {
        title, description, categoryID, itemID,
      } = this.state;
      // Set Item for comparing with current Item from props
      const item = {
        itemID,
        title,
        description,
        categoryID,
      };
      if (title.trim() === '') {
        errMessage('Title must not be blank');
        return;
      }
      if (description.trim() === '') {
        errMessage('Description must not be blank');
        return;
      }

      await this.props.editItem(item)
        .then(message => successMessage(message))
        .catch(err => errMessage(err));
      await this.props.getUserItems()
        .catch(err => errMessage(err));
      this.props.hideModal();
    }

    render() {
      const { modal, hideModal, categories } = this.props;
      const {
        title, description, categoryID,
      } = this.state;
      const isEditItemModal = modal === 'editItem';
      return (
        <div>
          <Modal isOpen={isEditItemModal} toggle={hideModal}>
            <ModalHeader toggle={hideModal}>Edit Item</ModalHeader>
            <ModalBody>
              <div>
                <label htmlFor="title">
                    Title:
                  <input type="text" name="title" value={title} onChange={this.changeInput} />
                </label>
              </div>
              <div>
                <label htmlFor="description">
                    Description:
                  <br />
                  <textarea name="description" value={description} onChange={this.changeInput} />
                </label>
              </div>
              <div>
                <label htmlFor="categoryID">
                    Category:
                  <select name="categoryID" value={categoryID} onChange={this.changeInput}>
                    {
                        categories.map(
                          ({ id, name }) => <option key={id} value={id}>{name}</option>,
                        )
                    }
                  </select>
                </label>
              </div>
            </ModalBody>
            <ModalFooter>
              <Button color="primary" onClick={this.editItem}>Edit</Button>
            </ModalFooter>
          </Modal>
        </div>
      );
    }
}

EditItemModal.propTypes = {
  modal: PropTypes.string.isRequired,
  item: PropTypes.object.isRequired,
  categories: PropTypes.array.isRequired,
  hideModal: PropTypes.func.isRequired,
  editItem: PropTypes.func.isRequired,
  getCategories: PropTypes.func.isRequired,
  getUserItems: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  modal: state.app.modal,
  item: state.user.currentItem,
  categories: selectAllCategories(state),
});

export default connect(mapStateToProps, {
  getCategories, getItems, getUserItems, editItem, hideModal,
})(EditItemModal);
