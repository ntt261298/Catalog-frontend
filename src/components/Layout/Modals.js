import React from 'react';
import Loadable from 'react-loadable';


export const LoginModal = Loadable({
  loader: () => import('components/Layout/LoginModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
export const SignupModal = Loadable({
  loader: () => import('components/Layout/SignupModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
export const AddItemModal = Loadable({
  loader: () => import('components/Layout/AddItemModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
export const EditItemModal = Loadable({
  loader: () => import('components/Layout/EditItemModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
