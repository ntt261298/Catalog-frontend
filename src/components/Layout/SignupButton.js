import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { showSignup } from '../../actions/app';

export const SignupButton = props => (
  <button type="button" onClick={props.showSignup} className="btn-signup">Signup</button>
);

SignupButton.propTypes = {
  showSignup: PropTypes.func.isRequired,
};

export default connect(null, { showSignup })(SignupButton);
