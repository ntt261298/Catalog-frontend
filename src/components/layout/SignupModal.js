import React from 'react';
import { connect } from 'react-redux';
import {
  Button, Modal, ModalHeader, ModalBody, ModalFooter,
} from 'reactstrap';
import { PropTypes } from 'prop-types';
import toastr from 'toastr';
import { showLogin, hideModal } from '../../actions/app';
import { onSignup } from '../../actions/user';


export class SignupModal extends React.Component {
  state = {
    username: '',
    password: '',
    message: '',
  }

  onChangeInput = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  moveToLogin = () => {
    this.props.hideModal();
    this.props.showLogin();
  }

  onSignup = async () => {
    const { username, password } = this.state;
    if (!username) {
      this.setState({
        message: 'Username must not be blank',
      });
      return;
    }
    if (password.length < 6) {
      this.setState({
        message: 'Password must have at least 6 characters',
      });
      return;
    }
    this.props.onSignup(this.state.username, this.state.password)
      .then((err) => {
        toastr.error('af');
      })
      .catch((err) => {
        toastr.error('af');
        // console.log(err);
      });
  }

  render() {
    const { modal, errMessage, hideModal } = this.props;
    const { message } = this.state;
    const isSignupModal = modal === 'signup';
    return (
      <div>
        <Modal isOpen={isSignupModal} toggle={hideModal}>
          <ModalHeader toggle={hideModal}>Signup</ModalHeader>
          <ModalBody>
            <div>
              <label htmlFor="username">
                Username:
                <input type="text" name="username" onChange={this.onChangeInput} />
              </label>
            </div>
            <div>
              <label htmlFor="password">
                Password:
                <input type="password" name="password" onChange={this.onChangeInput} />
              </label>
            </div>
            { message ? (
              <div className="alert alert-danger" role="alert">
                { message }
              </div>
            ) : null}
            { errMessage ? (
              <div className="alert alert-danger" role="alert">
                { errMessage }
              </div>
            ) : null}
            <p onClick={this.moveToLogin}>Sign in to your account?</p>
          </ModalBody>
          <ModalFooter>
            <Button color="primary" onClick={this.onSignup}>Signup</Button>
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

SignupModal.propTypes = {
  modal: PropTypes.string.isRequired,
  errMessage: PropTypes.string.isRequired,
  showLogin: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired,
  onSignup: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  modal: state.app.modal,
  errMessage: state.user.message,
});

export default connect(mapStateToProps, { showLogin, hideModal, onSignup })(SignupModal);
