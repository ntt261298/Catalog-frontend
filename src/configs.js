import {
  configure, shallow, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Enzyme config
configure({ adapter: new Adapter() });

// API host config
const defaultHost = 'http://localhost:5000';

export {
  shallow, mount, render, defaultHost,
};
