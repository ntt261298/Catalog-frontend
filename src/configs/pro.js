import base from 'configs/base';

export default {
  ...base,
};
