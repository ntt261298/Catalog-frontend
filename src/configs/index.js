import {
  configure, shallow, mount, render,
} from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import dev from 'configs/dev';
import pro from 'configs/pro';

// Enzyme config
configure({ adapter: new Adapter() });

let configs;

if (process.env.REACT_APP_ENV === 'dev') {
  configs = dev;
} else if (process.env.REACT_APP_ENV === 'pro') {
  console.log('pro');
  configs = pro;
} else {
  // Default configs is dev
  configs = dev;
}

export { shallow, mount, render };

export default configs;
