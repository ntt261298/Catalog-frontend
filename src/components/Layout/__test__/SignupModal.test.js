import React from 'react';
import { shallow } from 'configs';
import { SignupModal } from 'components/Layout/SignupModal';

describe('components/Layout/SignupModal', () => {
  let wrapper;
  let props;
  let signupButton;
  let moveToLoginParam;

  const update = () => {
    signupButton = wrapper.find('Button');
    moveToLoginParam = wrapper.find('p');
  };

  const setup = () => {
    wrapper = shallow(
      <SignupModal {...props} />,
    );
    update();
  };

  beforeEach(() => {
    props = {
      modal: 'signup',
      errMessage: '',
      onSignup: jest.fn().mockImplementation(() => Promise.resolve('message')),
      showLogin: jest.fn(),
      hideModal: jest.fn(),
    };
  });

  it('should render signup modal', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onSignup when signup button is clicked', () => {
    setup();
    wrapper.setState({ username: 'test', password: '123456' }, () => {
      signupButton.simulate('click');
      expect(props.onSignup).toHaveBeenCalled();
    });
  });

  it('should call showLogin and hideModal when new move to login param is clicked', () => {
    setup();
    moveToLoginParam.simulate('click');
    expect(props.hideModal).toHaveBeenCalled();
    expect(props.showLogin).toHaveBeenCalled();
  });

  it('should set message \'Username must not be blank\' when User\'s input is blank', () => {
    setup();
    wrapper.setState({ username: '', password: '123456' }, () => {
      signupButton.simulate('click');
      expect(props.onSignup).not.toHaveBeenCalled();
    });
  });

  it('should set message \'Password must have at least 6 characters\' when password\'s input is wrong', () => {
    setup();
    wrapper.setState({ username: 'test123', password: '123' }, () => {
      signupButton.simulate('click');
      expect(props.onSignup).not.toHaveBeenCalled();
    });
  });
});
