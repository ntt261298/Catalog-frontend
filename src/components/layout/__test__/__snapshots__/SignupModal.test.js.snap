// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/layout/SignupModal.js should render signup modal 1`] = `
<div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Signup
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <div>
        <label
          htmlFor="username"
        >
          Username:
          <input
            name="username"
            onChange={[Function]}
            type="text"
          />
        </label>
      </div>
      <div>
        <label
          htmlFor="password"
        >
          Password:
          <input
            name="password"
            onChange={[Function]}
            type="password"
          />
        </label>
      </div>
      <p
        onClick={[Function]}
      >
        Sign in to your account?
      </p>
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        onClick={[Function]}
        tag="button"
      >
        Signup
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
